daemon/src/wire.rs

```diff
@@ -50,8 +50,32 @@ pub type Write<D, E> = SplitSink<Framed<TcpStream, EncryptedJsonCodec<D, E>>, E>
 pub struct Version(semver::Version);
 
 impl Version {
+    const CURRENT: semver::Version = semver::Version::new(2, 0, 0);
+
     pub fn current() -> Self {
-        Self(semver::Version::new(2, 0, 0))
+        Self(Self::CURRENT)
+    }
+
+    pub fn is_compatible(taker: &Self, maker: &Self) -> bool {
```

Comment:
This could be also a method that could be used like this:

`our_version.is_compatible_with(&taker_version)` 

I'm not sure whether that would be better though (perhaps a bit more discoverable in the IDE?)

Reply:
That's what I had initially and writing the tests became more cumbersome so I opted for this one.